lib/attributor/types/hash.rb

```diff
@@ -418,6 +422,9 @@ def key?(k)
     end
     alias_method :has_key?, :key?
 
+    def merge(h)
+      self.class.load(@contents.merge(h))
```

Comment:
You might want to load the incoming `h`. Dunno if you need to load the full result though.